src/sql/base/browser/ui/modal/optionsDialog.ts

```diff
@@ -225,47 +221,51 @@ export class OptionsDialog extends Modal {
 
 	public open(options: sqlops.ServiceOption[], optionValues: { [name: string]: any }) {
 		this._optionValues = optionValues;
-		var firstOption: string;
-		var containerGroup: Builder;
-		var layoutSize = 0;
-		var optionsContentBuilder: Builder = $().div({ class: 'optionsDialog-options-groups' }, (container) => {
+		let firstOption: string;
+		let containerGroup: Builder;
+		let optionsContentBuilder: Builder = $().div({ class: 'optionsDialog-options-groups monaco-panel-view' }, (container) => {
 			containerGroup = container;
 			this._optionGroups = container.getHTMLElement();
 		});
-		var splitview = new SplitView(containerGroup.getHTMLElement());
+		this.splitview = new ScrollableSplitView(containerGroup.getHTMLElement(), { enableResizing: false, scrollDebounce: 0 });
 		let categoryMap = OptionsDialogHelper.groupOptionsByCategory(options);
-		for (var category in categoryMap) {
-			var serviceOptions: sqlops.ServiceOption[] = categoryMap[category];
-			var bodyContainer = $().element('table', { class: 'optionsDialog-table' }, (tableContainer: Builder) => {
+		for (let category in categoryMap) {
+			let serviceOptions: sqlops.ServiceOption[] = categoryMap[category];
+			let bodyContainer = $().element('table', { class: 'optionsDialog-table' }, (tableContainer: Builder) => {
 				this.fillInOptions(tableContainer, serviceOptions);
 			});
 
-			var viewSize = this._optionCategoryPadding + serviceOptions.length * this._optionRowSize;
-			layoutSize += (viewSize + this._categoryHeaderSize);
-			var categoryView = new CategoryView(category, bodyContainer.getHTMLElement(), false, viewSize, this._categoryHeaderSize);
-			splitview.addView(categoryView);
+			let viewSize = this._optionCategoryPadding + serviceOptions.length * this._optionRowSize;
+			let categoryView = this._instantiationService.createInstance(CategoryView, bodyContainer.getHTMLElement(), viewSize, { title: category, ariaHeaderLabel: category, id: category });
+			this.splitview.addView(categoryView, viewSize);
+			categoryView.render();
+			attachPanelStyler(categoryView, this._themeService);
 
 			if (!firstOption) {
 				firstOption = serviceOptions[0].name;
 			}
 		}
-		splitview.layout(layoutSize);
+		if (this.height) {
+			this.splitview.layout(this.height - 120);
```

Comment:
do we have to use the hard-coded size here and below? maybe use Flex container?

Reply:
You can't use flex containers for laying out ui elements programmatically since you would have to calculate the height of the flex container every time you call layout.